app/models/envelope.rb

```diff
@@ -60,25 +61,26 @@ class Envelope < ActiveRecord::Base
   NOT_FOUND = 'Envelope not found'.freeze
   DELETED = 'Envelope deleted'.freeze
 
-  def self.by_resource_custom_id(field, id)
-    find_by('processed_resource @> ?', { field => id }.to_json)
-  end
-
   def self.by_resource_id(id)
-    find_by('processed_resource @> ?', { '@id' => id }.to_json)
+    find_by 'lower(processed_resource::text)::jsonb @> ?',
```

Comment:
I believe this will cause a full table scan, also it will have to parse from all jsonb to text, then lowercase it all, then reconvert to jsonb, before running the matcher. 
Since we are using a lookup field, we can:
- preserve `by_resource_id` untouched. (or just add a lowercase index on it, like the previous solution from Max)
- on the lookup_id callback, you always store it lowercased
- on the query call downcase on the term.

Reply:
Yes, you're right about the performance issue here. This part is actually from Max's pull request so I assumed it would do the right thing (tm), but I see now that I took it from the older commit instead of the newer one, my mistake. I've replaced this with the newer code.

We do need to downcase inside `by_resource_id` however, and use a lowercase index, since fetching a resource by `@id` is the fallback if we don't find it via `lookup_id`, see here for the original logic we need to preserve: https://github.com/CredentialEngine/CredentialRegistry/blob/3bb53feca47ec87c61ade3964b290988896a799c/app/models/envelope.rb#L71

Fetching only by `lookup_id` would change that logic, and indeed it breaks existing tests.

As for storing the lookup_id lowercased, I considered it but decided to store it as is because this way we have the same logic for both `@id` and the new `lookup_id` instead of one of them using a lowercase index and the other using exact matching, it's easier on my brain :-).

Also it might make it easier to compare the contents of the lookup field with the original fields (@id, ceterms:ctid) at some point down the line if we need to. 

Another way would be to do what you said and store the lowercase `lookup_id`, and also lowercase the existing `@id`s in the database, and then simply compare directly with the downcased version of the id coming from the user, but I chose to leave the existing data untouched.

Let me know what you think.